BSP/Simulator/DataGeneration/simulate.py

```diff
@@ -213,10 +215,46 @@ def main():
                     print("Enter the frequency you would like to change the clock to in Hz.")
                     frequency = int(input())
                     PLL.Change_Frequency(frequency)
+                elif choice == 'EEPROM':
+                    errormsg = 0
+                    print("Enter 'all' for all data or 'read' to enter specific address to read.")
+                    print(">>", end="")
+                    choiceEEPROM1 = input()
+                    print("Enter 'raw' to read the raw hex values or 'msg' for the translated error messages.", end="\n")
+                    print("If invalid response is given, default is raw data.")
+                    choiceEEPROM2 = input()
+                    if choiceEEPROM2 == 'raw':
+                        errormsg = 0
+                    elif choiceEEPROM2 == 'msg':
+                        errormsg = 1
+                    else:
+                        print("Invalid entry...", end="\n")
+                        print("Defaulted to raw data.")
+                    if choiceEEPROM1 == 'all':
+                        print(I2C.EEPROM_Dump(errormsg))
+                        print("Enter to continue simulator:")
+                        print(">>", end="")
+                        choice = input()
+                    elif choiceEEPROM1 == 'read':
+                        print("Enter address to read faults from (in hex format).")
```

Comment:
The idea I had was for the user to be able to specify a range of EEPROM addresses that they want to read from (The addresses don't necessarily have to be faults). For example, the user might want to see the values of the pointers that are stored in the EEPROM that keep track of how large each array of faults in the EEPROM is.

Reply:
So do I need to implement an option to read from a given address to an end address & just spit out all the raw data between those two?

Reply:
Yes, that is what I had in mind.